### ISSUE TITLE
Pausing and resuming song seems to be broken.

### ISSUE THREAD

**Issue by joelbolz** [devgianlu/go-librespot#135]
When selecting the server go-librespot is running on as the device and playing a song, it works perfectly. I then can use the skip buttons both forwards and backwards as well as volume control. I can also tap on the "progress bar" of the song to skip ahead. Pause also works.
However, when trying to resume again, the audio does not start playing. On the spotify app, the Play/Pause button changes accordingly, but the playbackbar doesnt progress and there is no sound.

TLDR:

Steps to replicate:
Play song, pause song and press play again.

Result:
Song does not resume playing, playback bar first freezes, then skips to a random point but playback never actually resumes.

Expected result:
Song starts playing normally from the point it was paused at.

Info:
I am using PulseAudio, as I couldnt get the Bluetooth speaker to work with ALSA (not a go-librespot specific problem, couldnt get it to work with spocon either)

**Comment by joelbolz** [comment]
I just noticed: on podcasts, playing after pausing actually (kind of) works: it just plays it from the very beginning of the episode instead of where it was paused.

**Comment by devgianlu** [comment]
Thank you for reporting, there is temporary fix with #136, but I am waiting feedback on the best approach to fix this.

**Comment by aykevl** [comment]
> I am using PulseAudio, as I couldnt get the Bluetooth speaker to work with ALSA (not a go-librespot specific problem, couldnt get it to work with spocon either)

You mean you use the ALSA backend (the default) with PulseAudio?
Asking, because recently native PulseAudio support got merged: https://github.com/devgianlu/go-librespot/pull/86 - it will only be enabled if you have a very recent go-librespot build and have manually set `audio_backend: pulseaudio` in your configuration. Using the ALSA backend with PulseAudio should work just fine, but that PR could have introduced a regression in the ALSA backend too.

**Comment by joelbolz** [comment]
> Thank you for reporting, there is temporary fix with #136, but I am waiting feedback on the best approach to fix this.

Cool that's great to hear thank you man!

**Comment by joelbolz** [comment]
> > I am using PulseAudio, as I couldnt get the Bluetooth speaker to work with ALSA (not a go-librespot specific problem, couldnt get it to work with spocon either)
> 
> You mean you use the ALSA backend (the default) with PulseAudio? Asking, because recently native PulseAudio support got merged: #86 - it will only be enabled if you have a very recent go-librespot build and have manually set `audio_backend: pulseaudio` in your configuration. Using the ALSA backend with PulseAudio should work just fine, but that PR could have introduced a regression in the ALSA backend too.

No ALSA should be disabled. To be fair I'm not really super aware of what's going on with Pulseaudio and Alsa etc. But I'm pretty sure the Pulseaudio sinks are being used.
